### PR TITLE
Added natural logarithm algorithm

### LINKED ISSUE #4034: Implementing Natural logarithm algorithm
## 🚀 Feature

* Demonstrate a basic algorithm that some calculators use to compute natural logarithm. ( ln(x) )
* Script will not use any external libraries.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

- [x] Yes

## Motivation

I'm a GSSOC participant who is interested in many aspects of Mathematics including automating many of its functions using Python.

## Pitch

This script will give a general overview on how natural algorithm is dependent on Inverse Hyperbolic Functions and how some scientifc calculators use it to compute results.

### PR DESCRIPTION
**Pull request by XZANATOL**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

- [x] Yes

### Description

Added natural logarithm (ln) algorithm using Taylor's series along with the pros & cons of using this approach.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes: #4034

### CODE REVIEW

**Comment by XZANATOL** [comment]
@HarshCasper I don't know wether the resolve i did to resolve the merge conficts will affect the repository or not.. but if it will, let me know so that i reopen another PR instead of this one. 🤔

File: Python/README.md [hunk 1]
```diff
@@ -251,8 +251,12 @@
 - [Arctan Function](math/Arctan.py)
 - [Strong Number](math/Strong_Number.py)
 - [Noble Integer](math/Noble_Integer.py)
+<<<<<<< HEAD
```
Comment (ankitaggarwal23) [inline Python/README.md:254]: Remove that redundant line

File: Python/README.md [hunk 2]
```diff
@@ -251,8 +251,12 @@
...
+- [Natural Logarithm](math/Natural_Logarithm_Algorithm.py)
+=======
```
Comment (ankitaggarwal23) [inline Python/README.md:256]: Remove that redundant line

File: Python/README.md [hunk 3]
```diff
@@ -251,8 +251,12 @@
...
 - [Happy Number](math/Happy_Number.py)
 - [Magic Number](math/Magic_Number.py)
+>>>>>>> 53f43cf279f04ab5f30212c23fd06935ced40093
```
Comment (ankitaggarwal23) [inline Python/README.md:259]: Remove that redundant line

File: Python/math/Natural_Logarithm_Algorithm.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
+# Logarithm Algorithm (ln)
+# Written by XZANATOL
```
Comment (ankitaggarwal23) [inline Python/math/Natural_Logarithm_Algorithm.py:2]: Remove that redundant line

File: Python/math/Natural_Logarithm_Algorithm.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+"""
+An algorithm that demonstrates how to approximately calculate the \
+natural algorithm of a value and show how this logarithm is \
+dependent on Inverse hyperbolic functions.
+"""
+
+
```
Comment (ankitaggarwal23) [inline Python/math/Natural_Logarithm_Algorithm.py:10]: Remove extra one line

File: Python/math/Natural_Logarithm_Algorithm.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    """Returns abs of x if it's negative"""
+    if x < 0:
+        return abs(x), True
+    return x, False
+
+
+# Inverse Hyperbolic Functions return angles in radian form
+def artanh(val):
+    """Returns Inverse Hyperbolic Tan of x"""
+    val, neg_flag = conv_if_neg(val)  # artanh(-x) = -1 * artanh(x)
```
Comment (ankitaggarwal23) [inline Python/math/Natural_Logarithm_Algorithm.py:21]: Shift comment above the line of code

File: Python/math/Natural_Logarithm_Algorithm.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        return "Domain error" 
+    # Taylor series
+    res = 0
+    for i in range(1, 92, 2):
+        res += val**i/i
+    if neg_flag:  # Check negative flag
+        res *= -1
+    return res  # Radian angle
+
+
```
Comment (ankitaggarwal23) [inline Python/math/Natural_Logarithm_Algorithm.py:32]: Remove one line

File: Python/math/Natural_Logarithm_Algorithm.py [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+def ln(x):
+    """Returns the natural algorithm of x"""
+    # Use fact: ln(x) = 2 * artanh((x-1) / (x+1))
+    x = (x-1) / (x+1)
+    return 2 * artanh(x)
+
+
```
Comment (ankitaggarwal23) [inline Python/math/Natural_Logarithm_Algorithm.py:39]: Remove one line

File: Python/math/Natural_Logarithm_Algorithm.py [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+cannot be implemented without the help of the C language.
+Refer to function: "ln_values" in the following link:
+https://people.math.sc.edu/Burkardt/c_src/cordic/cordic.c
+
+Steps:
+=======
+It all depends on some expansion facts which are:
+    1) ln(x) = 2 * artanh((x-1) / (x+1))
+    2) artanh(x) = val + val**3/3 + val**5/5 + ...
+    3) artanh(-x) = -1 * artanh(x)
```
Comment (ankitaggarwal23) [inline Python/math/Natural_Logarithm_Algorithm.py:68]: Shift your approach after description

File: Python/math/Natural_Logarithm_Algorithm.py [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+Output:
+1.6094379124341003
+=============
+
+Time Complexity: O(1)
+Explain: Accuracy here is controlled by the developer so if the program was built on using \
+         using 300 Taylor terms, then the program will have a fixed time complexity of the \
+         time needed to calculate these 300 terms.
+"""
```
Comment (ankitaggarwal23) [inline Python/math/Natural_Logarithm_Algorithm.py:83]: Add SC as well.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by XZANATOL** [comment]
I've done the required changes on a second PR, as I had unresolvable problems with that one. Please continue with the review on PR #4418

**Pull request closed without merging**